Makefile

```diff
@@ -117,8 +97,52 @@ run-test-db:
   ./scripts/neo4j-wait-for-start;
 	make init-db
 
-clean-deps: unprepublish
-	rm -rf packages/*/node_modules
-	rm -rf node_modules
-	rm package-lock.json
-	npm install
+test:
+	@if [ -z $(CI) ]; \
+		then NEO4J_BOLT_URL=${LOCAL_BOLT_URL} TREECREEPER_TEST=true TREECREEPER_SCHEMA_DIRECTORY=example-schema DEBUG=true TIMEOUT=500000 \
+			jest --config="./jest.config.js" "${pkg}.*__tests__.*${spec}.*.spec.js" --testEnvironment=node --watch; \
+		else NEO4J_BOLT_URL=${LOCAL_BOLT_URL} TREECREEPER_TEST=true TREECREEPER_SCHEMA_DIRECTORY=example-schema \
+			jest --config="./jest.config.js" "__tests__.*/*.spec.js" --testEnvironment=node --maxWorkers=1 --ci --reporters=default --reporters=jest-junit --detectOpenHandles --forceExit; \
+	fi
+
+## Cypress stuff used in CI
+
+### Checks that cypress will be able to run ok
+cypress-verify:
+	TREECREEPER_TEST=true TREECREEPER_SCHEMA_DIRECTORY=example-schema \
+	cypress verify
+
+### Runs tests for pages
+cypress-page: build-statics
+	start-server-and-test "make run-app" http-get://localhost:8888/MainType/create "make cypress-run-page"
+
+### Runs tests for primitive components
+cypress-primitives: build-statics
+	start-server-and-test "make run-app" http-get://localhost:8888/MainType/create "make cypress-run-primitives"
+
+## Cypress stuff used in local dev
+
+### Opens an interactive UI for specifying which tests to run/re-run
+cypress-open:
+	TREECREEPER_TEST=true TREECREEPER_SCHEMA_DIRECTORY=example-schema \
+	cypress open
+
+### Runs tests for pages (assumes the app is running in another terminal)
+cypress-run-page:
+	TREECREEPER_TEST=true TREECREEPER_SCHEMA_DIRECTORY=example-schema \
+	cypress run --spec "packages/tc-ui/src/pages/**/__tests__/**.cyp.js"
+
+### Runs tests for primitive components (assumes the app is running in another terminal)
+cypress-run-primitives:
+	TREECREEPER_TEST=true TREECREEPER_SCHEMA_DIRECTORY=example-schema \
+	cypress run --spec "packages/tc-ui/src/primitives/**/__tests__/**.cyp.js"
+
+# Deploy
+
+prepublish:
+	babel packages/tc-ui/src -D --out-dir packages/tc-ui/dist
+	sed s/"src\/"/"dist\/"/ packages/tc-ui/package.json > tmp && mv tmp packages/tc-ui/package.json
```

Comment:
Should this be a `prepare` script within the `tc-ui` package itself?

Reply:
Probably, but there might be a good reason to keep separate... something something... local dev... mumble mumble

Will leave as is for now

Reply:
I did see that this tweaks the entry point of the package manifest - I assume so local linking can consume from source instead of rebuilding each change? One for a separate PR anyway.